prereq_flowchart/compose.py

```diff
@@ -0,0 +1,67 @@
+from os.path import join
+from hashlib import md5
+
+from prereq_flowchart.scrape.classinfo import *
+from prereq_flowchart.scrape import scrape
+
+"""
+recursively searches for prerequisites to a class, then flattens it into a list
+of courses and their prerequisites.
+
+this is the memoization test
+"""
+# def get_prerequisites():
+
+
+def make_json_data_for_majors():
+    c = scrape.scrape_classinfo()
+    m = scrape.scrape_majors()
+    # associate hashed file name with major title
+    major_table = {}
+
+    for major in m:
+        print(f"generating: {major.name}")
+        courses = {}
+        for course in major.courses:
+            num = course.catalogTitle
+
+            # i don't know how to do this better, it's painfully slow
+            # creating a CourseNumber and checking for equality does not work
```

Comment:
on the definition of `class CourseNumber`, change the `@dataclass(frozen=True)` decorator to be `@dataclass(eq=True, frozen=True)`. 

the `eq=True` part will let you use `==` on `CourseNumber`s

the `frozen=True` part (in conjuction with `eq=True`) will allow python to generate a `__hash__` method for `CourseNumber`, which lets you use it as dict keys/set values